### PR TITLE
ci: Run tests also against CentOS 9 (OpenSSL 3.0)

### PR DESCRIPTION
**Pull request by Jakuje**
#### Description

The Debian is unstable so it is moving quite fast with the OpenSSL versions. Fedora will update from OpenSSL 3.1 to 3.2 in coming month so we will not be able to check the changes keep working with older OpenSSL versions, which is our interest for RHEL9.

#### Checklist

- [X] Code modified for feature

#### Reviewer's checklist:

- [x] Any issues marked for closing are addressed
- [x] There is a test suite reasonably covering new functionality or modifications
- [x] This feature/change has adequate documentation added
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [x] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,19 +14,21 @@ jobs:
...
           - name: debian
             container: debian:sid
+          - name: centos
+            container: quay.io/centos/centos:stream9
     container: ${{ matrix.container }}
     steps:
       - name: Install Dependencies
         run: |
-            if [ -f /etc/fedora-release ]; then
+            if [ -f /etc/redhat-release ]; then
```
Comment (simo5) [inline .github/workflows/build.yml:31]: we should probably use /etc/system-release here?

**Review by simo5 — APPROVED** [review]
LGTM, let me know if you want to change the code to use the system-release file, I do not think it is a deal breaker, it works as is as well.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Jakuje) [reply .github/workflows/build.yml:31]: I do not have preference. If you believe the `system-release` is more appropriate, I can change it. I see its not in debian containers. But I thought the `redhat-release` was more self-documented.

Comment (simo5) [reply .github/workflows/build.yml:31]: if debian does not have it no real need then.

**Pull request merged**